sdk/eventhub/event-hubs/src/batchingReceiver.ts

```diff
@@ -167,6 +182,7 @@ export class BatchingReceiver extends EventHubReceiver {
       };
 
       onReceiveClose = async (context: EventContext) => {
+        this.isReceivingMessages = false;
```

Comment:
Let's add this is `onReceiverError` and `onSessionError` as well.
We did discuss offline that the receiver and session `close` events are always fired after the corresponding `error` events, and so decided to have this only in the `close` event handlers, but take the below case
- User calls receiveBatch
- There is a receiver error and rhea fires the `error` event and then the `close` event
- Receiver error handler is called which rejects the promise returned by receiveBatch
- Now, I am not sure if the Receiver close handler is called before the next line in the user's code is called. If the user's code is called next, and they check `isReceivingMessages`, it will wrongly return `true`

So, to be safe, lets set `isReceivingMessages` to `false` in both the `error` and `close` event handlers for receivers and sessions

Reply:
Updated!!